src/views/Farms/components/FarmCard/StakeAction.tsx

```diff
@@ -36,6 +39,18 @@ const StakeAction: React.FC<FarmCardActionsProps> = ({
   const { onStake } = useStake(pid)
   const { onUnstake } = useUnstake(pid)
   const location = useLocation()
+  const dispatch = useAppDispatch()
+  const { account } = useWeb3React()
+
+  const handleStake = async (amount: string) => {
+    await onStake(amount)
+    dispatch(fetchFarmUserDataAsync({ account, pids: [pid] }))
```

Comment:
This is where we refresh the user data (originally it was in `onStake` and was refreshing all user data)